components/camel-minio/src/main/java/org/apache/camel/component/minio/MinioProducer.java

```diff
@@ -0,0 +1,494 @@
+        }
+
+        String contentDisposition = exchange.getIn().getHeader(MinioConstants.CONTENT_DISPOSITION, String.class);
+        if (contentDisposition != null) {
+            objectMetadata.put("Content-Disposition", contentDisposition);
+        }
+
+        String contentEncoding = exchange.getIn().getHeader(MinioConstants.CONTENT_ENCODING, String.class);
+        if (contentEncoding != null) {
+            objectMetadata.put("Content-Encoding", contentEncoding);
+        }
+
+        String contentMD5 = exchange.getIn().getHeader(MinioConstants.CONTENT_MD5, String.class);
+        if (contentMD5 != null) {
+            objectMetadata.put("Content-Md5", contentMD5);
+        }
+
+        return objectMetadata;
+    }
+
+    /**
+     * Reads the bucket name from the header of the given exchange. If not
+     * provided, it's read from the endpoint configuration.
+     *
+     * @param exchange The exchange to read the header from.
+     * @return The bucket name.
+     * @throws IllegalArgumentException if the header could not be determined.
+     */
+    private String determineBucketName(final Exchange exchange) {
+        String bucketName = exchange.getIn().getHeader(MinioConstants.BUCKET_NAME, String.class);
+
+        if (ObjectHelper.isEmpty(bucketName)) {
+            bucketName = getConfiguration().getBucketName();
+            LOG.trace("Minio Bucket name header is missing, using default one [{}]", bucketName);
+        }
+
+        if (bucketName == null) {
+            throw new IllegalArgumentException("Minio Bucket name header is missing or not configured.");
+        }
+
+        return bucketName;
+    }
+
+    private String determineObjectName(final Exchange exchange) {
+        String objectName = exchange.getIn().getHeader(MinioConstants.OBJECT_NAME, String.class);
+        if (ObjectHelper.isEmpty(objectName)) {
+            objectName = getConfiguration().getKeyName();
+        }
+        if (objectName == null) {
+            throw new IllegalArgumentException("Minio Key header is missing.");
+        }
+        return objectName;
+    }
+
+    private String determineStorageClass(final Exchange exchange) {
+        String storageClass = exchange.getIn().getHeader(MinioConstants.STORAGE_CLASS, String.class);
+        if (storageClass == null) {
+            storageClass = getConfiguration().getStorageClass();
+        }
+
+        return storageClass;
+    }
+
+    private String determineVersionId(final Exchange exchange) {
+        String versionId = exchange.getIn().getHeader(MinioConstants.VERSION_ID, String.class);
+        if (versionId == null) {
+            versionId = getConfiguration().getVersionId();
+        }
+
+        return versionId;
+    }
+
+    private ByteArrayOutputStream determineLengthInputStream(InputStream inputStream) throws IOException {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        byte[] bytes = new byte[1024];
```

Comment:
why is it `1024` bits here? Can you please move this into constant and with some comment why you opted for `1024`bits?

Reply:
Hi @omarsmak, I was actually mimicking [this](https://github.com/apache/camel/blob/d4cfbd3631a0b277b5228224e1855c7229062c21/components/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/AWS2S3Producer.java#L573) line. I will put this as a constant in MinioConstants.java

Reply:
Hi @omarsmak, I was actually mimicking [this](https://github.com/apache/camel/blob/d4cfbd3631a0b277b5228224e1855c7229062c21/components/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/AWS2S3Producer.java#L573) line. I will put this as a constant in MinioConstants.java